zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/SparkSQLGenerator.scala

```diff
@@ -0,0 +1,636 @@
+              if (field.name == "create_at"){
+                //                producedExprs +=
+              }
+              else {
+                producedExprs += field.name -> exprMap(field.name)
+              }
+          }
+        }
+        val newExprMap = producedExprs.result().toMap
+        println("newExprMap:", newExprMap)
+        //check if fields is empty
+        val projectStr = if (select.fields.isEmpty) {
+          //          if (query.hasUnnest || query.hasGroup) {
+          //            parseProject(exprMap)
+          //          } else {
+          //            s"select *"
+          //          }
+          s"select *"
+        } else {
+          //          parseProject(newExprMap)
+          parseProject(exprMap)
+        }
+        queryBuilder.insert(0, projectStr + "\n")
+        ParsedResult(Seq.empty, newExprMap)
+
+      case None =>
+        val projectStr =
+          if (query.hasUnnest || query.hasGroup) {
+            parseProject(exprMap)
+          } else {
+            s"select *"
+          }
+        queryBuilder.insert(0, projectStr + "\n")
+        ParsedResult(Seq.empty, exprMap)
+    }
+  }
+
+  private def parseProject(exprMap: Map[String, FieldExpr]): String = {
+    exprMap.filter {
+      case (field, expr) => field != "*" && expr.refExpr != sourceVar
+    }.map {
+      case (field, expr) =>
+        if (s"${expr.defExpr}" == "`hour`" || s"${expr.defExpr}" == "`day`" || s"${expr.defExpr}" == "`month`"){
+          //          s"${expr.defExpr}(t.`create_at`) as $quote$field$quote"
+          s"${expr.defExpr}(t.`create_at`)"
+        }
+        else {
+          s"${expr.defExpr} as $quote$field$quote"
+        }
+    }.mkString("select ", ",", "")
+  }
+
+  private def parseGlobalAggr(globalAggrOpt: Option[GlobalAggregateStatement],
+                              exprMap: Map[String, FieldExpr],
+                              queryBuilder: StringBuilder): ParsedResult = {
+    globalAggrOpt match {
+      case Some(globalAggr) =>
+        val producedExprs = mutable.LinkedHashMap.newBuilder[String, FieldExpr]
+        val aggr = globalAggr.aggregate
+        val funcName = getAggregateStr(aggr.func)
+
+        val newDefExpr = if (aggr.func == Count) {
+          globalAggrVar
+        } else {
+          s"${sourceVar}.$quote${aggr.field.name}$quote"
+        }
+        val newRefExpr = s"$quote${aggr.as.name}$quote"
+
+        producedExprs += aggr.as.name -> FieldExpr(newRefExpr, newDefExpr)
+        val prepend =
+          s"""
+             |select $funcName($newDefExpr) as $quote${aggr.as.name}$quote from
+             |(""".stripMargin
+        val append =
+          s""")""".stripMargin
+        queryBuilder.insert(0, prepend)
+        queryBuilder.append(append)
+        ParsedResult(Seq.empty, producedExprs.result().toMap)
+      case None =>
+        ParsedResult(Seq.empty, exprMap)
+    }
+  }
+
+  protected def parseNumberRelation(filter: FilterStatement,
```

Comment:
Why this method down here? You should reorder them to make related methods come together.